modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp

```diff
@@ -0,0 +1,323 @@
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/fieldlinessequence/rendering/renderablefieldlinessequence.h>
+
+#include <openspace/engine/openspaceengine.h>
+#include <openspace/rendering/renderengine.h>
+#include <openspace/util/updatestructures.h>
+
+#include <ghoul/opengl/programobject.h>
+#include <ghoul/opengl/textureunit.h>
+
+
+namespace {
+    std::string _loggerCat = "RenderableFieldlinesSequence";
+
+    const GLuint _VA_POSITION = 0; // MUST CORRESPOND TO THE SHADER PROGRAM
+    const GLuint _VA_COLOR    = 1; // MUST CORRESPOND TO THE SHADER PROGRAM
+    const GLuint _VA_MASKING  = 2; // MUST CORRESPOND TO THE SHADER PROGRAM
+} // namespace
+
+namespace openspace {
+
+void RenderableFieldlinesSequence::deinitialize() {
+    glDeleteVertexArrays(1, &_vertexArrayObject);
+    _vertexArrayObject = 0;
+
+    glDeleteBuffers(1, &_vertexPositionBuffer);
+    _vertexPositionBuffer = 0;
+
+    glDeleteBuffers(1, &_vertexColorBuffer);
+    _vertexColorBuffer = 0;
+
+    glDeleteBuffers(1, &_vertexMaskingBuffer);
+    _vertexMaskingBuffer = 0;
+
+    RenderEngine& renderEngine = OsEng.renderEngine();
+    if (_shaderProgram) {
+        renderEngine.removeRenderProgram(_shaderProgram);
+        _shaderProgram = nullptr;
+    }
+
+    // Stall main thread until thread that's loading states is done!
+    while (_isLoadingStateFromDisk) {
+        LWARNING("TRYING TO DESTROY CLASS WHEN A THREAD USING IT IS STILL ACTIVE");
```

Comment:
Looks like this could result in infinite printing?

Reply:
Yeah, it looks that way, but I don't think it will. 
The reason for that `while` loop is that I'm sometimes loading states on a different thread; if so the `_isLoadingStatesFromDisk` is set to `true` and as soon as the thread is done it is set back to `false`. In this other thread it makes use of some member variables, which is  why I must let that thread finish before the destrutor can continue.
Is there another way of achieving this functionality?

BTW: It is the function `RenderableFieldlinesSequence::readNewState(const std::string& FILEPATH)` that is being executed if `_isLoadingStatesFromDisk == true`

Reply:
Oh, btw: I have never seen that message being triggered, so it's something that is VERY unlikely to happen!

Reply:
I assume the `_isLoadingStateFromDisk` is `std::atomic_bool`  if so, there is no other good way of doing that sans using a full mutex, which I think would be overkill here

Reply:
Alright! I'll leave it be then.